docs/Caching.md

```diff
@@ -0,0 +1,158 @@
+responsiveness. There are two primary types of content that are cached:
+
+1. REST API responses
+2. Static assets (files)
+
+## Overview
+
+![Cache diagram](https://user-images.githubusercontent.com/1885153/27892985-d75546fe-6256-11e7-8272-66252caa5c85.png)
+
+([from web platform diagrams](https://docs.google.com/presentation/d/1c8jf8UtGa81Ay4oqlbHfoP4Ile2VkiRVrCMVag6cRBQ/#slide=id.g1f921bb8da_0_0))
+
+## 1. API responses
+
+Currently the Meetup REST API does not provide a
+[`Cache-Control` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)
+(e.g. `max-age`), so the 'raw' API responses cannot be directly cached at the
+app server or browser level - the app must re-fetch every API request from the 
+browser to ensure that it delivers the requested data.
+
+### CDN
+
+The CDN does not currently cache any API responses. [See the 'Future'
+discussion](#future) for potential changes to the behavior.
+
+### App server
+
+The app server does not currently cache any API responses. [See the 'Future' 
+discussion](#future) for potential changes to the behavior.
+
+### Browser
+
+The browser uses
+[IndexedDB](https://developer.mozilla.org/en/docs/Web/API/IndexedDB_API) for
+local storage of API response data - it caches the JSON response rather than the
+full HTTP response in order to avoid complexity arising from the batched API
+request/response routine.
+
+#### Service Worker
+
+The service worker does not cache API requests/responses (and probably
+shouldn't - [see the 'Future' discussion](#future)).
+
+#### Cache middleware
+
+The cache epic (middleware) provides optimistic `state` updates for any data
+specified by dispatched `GET` queries, similar to the Sync epic, but uses
+locally cached data rather than an HTTP response. It is client-side only, and it
+*does not suppress* data fetched from the server - it simply pre-empts it before
+the API response returns and overwrites everything with the latest server data.
+[See the 'Future' discussion](#future) for potential changes to this behavior.
+
+##### Cache lifecycle
+
+**on `API_REQ`**, which provides `queries` to be sent to the API:
+
+1. For each query, use the JSON-encoded query as a key into the cache
+2. Format responses from the cache to look like API responses
+3. Trigger `CACHE_SUCCESS` containing the cache hits
+
+**on `API_SUCCESS`**,  which provides fresh info from the server
+
+- make an entry in the cache with `key` as the JSON-encoded `query` and
+	the `value` as the corresponding `response`
+
+The cache is stored locally in the user's browser using IndexedDB so that it
+will survive multiple sessions. All API queries are stored, and the data is
+refreshed each time the query is re-requested.
+
+In dev, be aware that the cache may be masking failed API requests - keep your
+network dev tools open and watch the server logs to make sure you haven't broken
+anything.
+
+### Future
```

Comment:
I'm a little confused about including the Future section here, checked into the repo. Seems like this is more 'Issues' territory.

I'd like to have some conversation about which options we may consider in the future but would like to find another venue to discuss it.